core/src/main/java/org/apache/calcite/avatica/remote/looker/LookerRemoteMeta.java

```diff
@@ -0,0 +1,375 @@
+      connection.setHostnameVerifier(trustAllHostNames);
+    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * A regrettable necessity. The Kotlin SDK relies on an outdated Ktor HTTP client based on Kotlin
+   * Coroutines which are a tad difficult to work with in Java. Here we make a HTTP client to handle
+   * the response stream ourselves which is "good enough" for now. This method should be revisited
+   * when the Kotlin SDK has built-in streams.
+   *
+   * TODO https://github.com/looker-open-source/sdk-codegen/issues/1341:
+   *  Add streaming support to the Kotlin SDK.
+   */
+  private void streamResponse(String url, OutputStream outputStream) throws IOException {
+    // use some SDK client helpers to tighten up this call
+    AuthSession authSession = getSdk().getAuthSession();
+    Transport sdkTransport = authSession.getTransport();
+    // makes a proper URL from the API endpoint path as the SDK would.
+    String endpoint = sdkTransport.makeUrl(url, Collections.emptyMap(), null);
+    URL httpsUrl = new URL(endpoint);
+    HttpsURLConnection connection = (HttpsURLConnection) httpsUrl.openConnection();
+    // WARNING: You should only set `verifySSL=false` for local/dev instances!!
+    if (!sdkTransport.getOptions().getVerifySSL()) {
+      trustAllHosts(connection);
+    }
+    connection.setRequestMethod("POST");
+    connection.setDoOutput(true);
+    // Set the auth header as the SDK would
+    connection.setRequestProperty("Authorization",
+        "token " + authSession.getAuthToken().getAccessToken());
+    connection.setRequestProperty("Accept", "application/json");
+    int responseCode = connection.getResponseCode();
+    if (responseCode == 200) {
+      // grab the input stream and write to the output for the main thread to consume.
+      InputStream inputStream = connection.getInputStream();
+      int bytesRead;
+      byte[] buffer = new byte[DEFAULT_STREAM_BUFFER_SIZE];
+      while ((bytesRead = inputStream.read(buffer)) != -1) {
+        outputStream.write(buffer, 0, bytesRead);
+      }
+      outputStream.close();
+      inputStream.close();
+    } else {
+      throw new IOException("HTTP request failed with status code: " + responseCode);
+    }
+  }
+
+  /**
+   * Prepares a thread to stream query response into an OutputStream. Sets an exception handler for
+   * the thread for reporting.
+   */
+  private Thread streamingThread(String baseUrl, InputStream in, OutputStream out) {
+    Thread stream = new Thread(() -> {
+      try {
+        streamResponse(baseUrl, out);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    });
+    Thread.UncaughtExceptionHandler exceptionHandler = (th, ex) -> {
+      try {
+        in.close();
+        out.close();
+        throw new RuntimeException(ex);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
The documented JVM behavior for exceptions thrown from the uncaught exception handler is to ignore them. You do get output on the terminal. The client code in `LookerIterable` is just going to get a pipe closed exception though.

Depending on how much implementation detail information you want to transmit from this thread to the client, it may be worthwhile marshaling exceptions from the streaming thread to the thread consuming the iterable, by e.g. sharing an atomic reference between the two, storing an exception before closing the pipe, and checking the atomic upon getting a pipe closed exception on the other side.

Of course the [JDK doesn't help](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/io/PipedInputStream.java#L266C50-L266C50) with a specific exception type for pipe closed, it's just a custom message inside an `IOException`. There are a few ways to solve this problem, exercise for the reader...